### PR TITLE
[Alberto Icardi] Sequence_task_1

### PR DESCRIPTION
**Pull request by AlbertoIcardi97**

### CODE REVIEW

File: sequence_task_1/student/include/impl.h [hunk 1]
```diff
@@ -3,39 +3,82 @@
...
 #include <stdexcept>
 
+
+
 template<class Container, class Song_t>
 class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    typedef Song_t value_type;
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: always prefer using over typedef

File: sequence_task_1/student/include/impl.h [hunk 2]
```diff
@@ -3,39 +3,82 @@
...
+    typedef Song_t& reference;
+    typedef const Song_t & const_reference;
+    typedef typename Container::iterator iterator;
+    typedef typename Container::const_iterator const_iterator;
+    typedef ptrdiff_t difference_type;
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: you can just reuse it from the container

File: sequence_task_1/student/include/impl.h [hunk 3]
```diff
@@ -3,39 +3,82 @@
...
-
+    template<class OtherContainer> StaticPlaylist & operator=(const OtherContainer & other){
+        m_tracklist.assign(other.rbegin(),other.rend());
+        return *this;
+    }
     /** @todo Add track from initializer */
     template<class... Args>
-    const Song_t& play(Args&&... songData);
+    const Song_t& play(Args&&... songData){
+        m_tracklist.emplace(end(),std::forward<Args>(songData)...);
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: emplace_back()?

File: sequence_task_1/student/include/impl.h [hunk 4]
```diff
@@ -3,39 +3,82 @@
...
+        return current();
+    }
 
     /** @todo Add track */
-    const Song_t& play(const Song_t& song);
+    const Song_t& play(const Song_t& song){
+        m_tracklist.insert(end(),song);
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: push_back?

File: sequence_task_1/student/include/impl.h [hunk 5]
```diff
@@ -3,39 +3,82 @@
...
+    const_iterator end() const{
+        return m_tracklist.cend();
+    };
 
     StaticPlaylist() = default;
 
     /** @todo Constructor from any reversible sequence container */
+ 
+    template<class OtherContainer> StaticPlaylist(const OtherContainer & tracklist){
+       m_tracklist.assign(tracklist.rbegin(),tracklist.rend());
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: can we use constructor of tracklist here?

Comment (AlbertoIcardi97) [reply sequence_task_1/student/include/impl.h]: Yes in this way: 
template<class OtherContainer> StaticPlaylist(const OtherContainer & tracklist):
    m_tracklist(tracklist.rbegin(),tracklist.rend()){}

**Pull request closed without merging**